src/Features/DCBreadcrumb.php

```diff
@@ -111,11 +141,85 @@ public function getDcBreadcrumb()
                         'linkoption' => $datacenter->isDeleted() ? 'class="target-deleted"' : '',
                         'icon'       => true
                     ];
-                    $breadcrumb[] = $datacenter->getLink($options);
+
+                    $breadcrumb_item = [
+                        'link' => $datacenter->getLink($options),
+                    ];
+
+                    if ($with_location) {
+                        $breadcrumb_item['location']['item'] = Location::getFromItem($datacenter);
+                    }
+
+                    $breadcrumb[Datacenter::getType()] = $breadcrumb_item;
                 }
             }
         }
 
+        $breadcrumb = array_reverse($breadcrumb);
+
+        if ($with_location) {
+            $breadcrumb = $this->cleanLocationInformations($breadcrumb);
+        }
+
+        return $breadcrumb;
+    }
+
+    /**
+     * Clean Location information foreach item breadcrumb
+     * If Current item Location = Parent item Location              -> do not display Location
+     * If Current item Location is child of Parent item Location    -> display only Location name instead of completename
+     * Else Current item Location != Parent item Location           -> display Location completename
+     *
+     * @param array $breadcrumb
+     *
+     * @return array cleanded $breadcrumb
+     */
+    public function cleanLocationInformations(&$breadcrumb)
```

Comment:
1. Your return the updated array, so no need to use a reference.
2. Method should be private as it is used only here.
```suggestion
    private function cleanLocationInformations(array $breadcrumb)
```